src/diffpy/utils/diffraction_objects.py

```diff
@@ -217,95 +217,48 @@ def __add__(self, other):
 
     __radd__ = __add__
 
+    def __sub__(self, other):
+        self._check_operation_compatibility(other)
+        subtracted_do = deepcopy(self)
+        if isinstance(other, (int, float)):
+            subtracted_do._all_arrays[:, 0] -= other
+        if isinstance(other, DiffractionObject):
+            subtracted_do._all_arrays[:, 0] -= other.all_arrays[:, 0]
+        return subtracted_do
+
+    __rsub__ = __sub__
+
+    def __mul__(self, other):
+        self._check_operation_compatibility(other)
+        multiplied_do = deepcopy(self)
+        if isinstance(other, (int, float)):
+            multiplied_do._all_arrays[:, 0] *= other
+        if isinstance(other, DiffractionObject):
+            multiplied_do._all_arrays[:, 0] *= other.all_arrays[:, 0]
+        return multiplied_do
+
+    __rmul__ = __mul__
+
+    def __truediv__(self, other):
+        self._check_operation_compatibility(other)
+        divided_do = deepcopy(self)
+        if isinstance(other, (int, float)):
+            divided_do._all_arrays[:, 0] /= other
+        if isinstance(other, DiffractionObject):
+            divided_do._all_arrays[:, 0] /= other.all_arrays[:, 0]
+        return divided_do
+
+    __rtruediv__ = __truediv__
+
     def _check_operation_compatibility(self, other):
         if not isinstance(other, (DiffractionObject, int, float)):
             raise TypeError(invalid_add_type_emsg)
         if isinstance(other, DiffractionObject):
             self_yarray = self.all_arrays[:, 0]
             other_yarray = other.all_arrays[:, 0]
-            if len(self_yarray) != len(other_yarray):
+            if self_yarray.shape != other_yarray.shape:
```

Comment:
@sbillinge as suggested - using `shape`

Reply:
the idea for using shape was to shorten code and make it more readable, so 
```
            self_yarray = self.all_arrays[:, 0]
            other_yarray = other.all_arrays[:, 0]
            if len(self_yarray) != len(other_yarray):
                raise ValueError(y_grid_length_mismatch_emsg)
```
is replaced by 
```
            if shape(self.all_arrays) != shape(other.all_arrays):
                raise ValueError(y_grid_length_mismatch_emsg)
```